src/cryptonote_core/blockchain.cpp

```diff
@@ -3477,6 +3479,7 @@ bool Blockchain::cleanup_handle_incoming_blocks(bool force_sync)
   CRITICAL_REGION_LOCAL(m_blockchain_lock);
   TIME_MEASURE_START(t1);
 
+  m_db->batch_stop();
```

Comment:
There will always be a batch active, unless block_entry.size() == 0. In that case there's nothing to clean up.